lib/adapter.js

```diff
@@ -4183,42 +4182,27 @@ function Adapter(options) {
                 if (err || !res || !res.rows) {
                     return void callback(err);
                 }
-                let count = 0;
-                for (let i = 0; i < res.rows.length; i++) {
-                    count++;
-                    adapterObjects.getObject(res.rows[i].id, options, (err, obj) => {
-                        if (err) {
-                            if (callback) {
-                                callback(err);
-                                callback = null;
-                            }
-                            return;
-                        } else if (obj && obj.common && obj.common.members) {
-                            const pos = obj.common.members.indexOf(objId);
-                            if (pos !== -1) {
-                                obj.common.members.splice(pos, 1);
-                                count++;
-                                obj.from = 'system.adapter.' + this.namespace;
-                                obj.user = (options ? options.user : '') || 'system.user.admin';
-                                obj.ts = Date.now();
-                                adapterObjects.setObject(obj._id, obj, err => {
-                                    if (!--count && callback) {
-                                        callback(err);
-                                        callback = null;
-                                    }
-                                });
+                const promises = [];
+                for (const row of res.rows) {
+                    promises.push(new Promise(resolve => {
+                        adapterObjects.getObject(row.id, options, (err, obj) => {
+                            if (err) {
+                                return typeof callback === 'function' && callback(err);
+                            } else if (obj && obj.common && obj.common.members) {
+                                const pos = obj.common.members.indexOf(objId);
+                                if (pos !== -1) {
+                                    obj.common.members.splice(pos, 1);
+                                    obj.from = 'system.adapter.' + this.namespace;
+                                    obj.user = (options ? options.user : '') || 'system.user.admin';
+                                    obj.ts = Date.now();
+                                    promises.push(new Promise(resolve => adapterObjects.setObject(obj._id, obj, resolve)));
```

Comment:
I'm not sure about the nested promise array pushing. How about an IIAFE (`(async () => { ... })();`) that just uses the async versions of get/set object and awaits them in a for-loop?

Reply:
good point wasn't aware of an existing `getObjectAsync` method in `objects` lib

Reply:
I discussed it with @Apollon77 yesterday. Both getObject and getObjectAsync exist, getObject additionally handles the case when no callback is passed and returns a Promise. Not sure right now if that is the case for FilesDB too.

Reply:
Yeah, wasn't sarcastic or anything like that..;-)
I really wasn't aware and saw it as you pointed out, we have it for files and for redis.. so yeah makes sense to use it here